File: day1/day1.jl
files = open("day1_input.txt", "r")

function load_file(files)
    left_list = []
    right_list = []

    for lines in readlines(files)
        line = split(lines)
        # Julia is 1 based Index
        push!(left_list, line[1])
        push!(right_list, line[2])
    end
    # Converting list from str to Int
    left = parse.(Int, left_list)
    right = parse.(Int, right_list)

    return left, right
end

function part_1(files)
    left, right = load_file(files)

    sort!(left)
    sort!(right)
    position_diff = []

    for (l,r) in zip(left, right)
        push!(position_diff, abs(l-r))
    end

    return sum(position_diff)
    
end

function part_2(files)
    left, right = load_file(files)

    similarity = []
    for ele in left
        count_element = count(x -> x == ele, right)
        s =  count_element*ele
        if s > 0
            push!(similarity,s)
        end
    end

    return sum(similarity)
end


println(part_1(files))
println(part_2(files))

File: day1/day1.py
def load_input(path:str)->list:

    files = open(path, 'r+')

    left_list = [line.strip().split()[0] for line in files]
    right_list = [line.strip().split()[1] for line in files]
    
    
    return left_list, right_list


def part_1(path:str)->int:
    
    left_list, right_list = load_input(path)

    left_list.sort()
    right_list.sort()

    location_diff = []

    for l,r in zip(left_list, right_list):
        diff = abs(int(l) - int(r))
        location_diff.append(diff)

    return sum(location_diff)

def part_2(path:str)->int:
    left_list, right_list = load_input(path)

    total = 0
    for ele in left_list:
        count = right_list.count(ele)
        total += count * int(ele)
    
    return total

#Doc Test
if __name__ == "__main__":

    print("***** Part 1 *****")
    print(part_1('day1_input.txt'))
    print("*"*18)

    print("***** Part 2 *****")
    print(part_2('day1_input.txt'))
    print("*"*18)